qiita_ware/ebi.py

```diff
@@ -482,53 +398,70 @@ def generate_experiment_xml(self):
             design = ET.SubElement(experiment, 'DESIGN')
             design_description = ET.SubElement(design,
                                                'DESIGN_DESCRIPTION')
-            design_description.text = escape(clean_whitespace(
-                sample_info['prep']['experiment_design_description']))
+            edd = sample_prep.pop('experiment_design_description')
+            design_description.text = escape(clean_whitespace(edd))
             ET.SubElement(
                 design, 'SAMPLE_DESCRIPTOR', {'refname': sample_alias}
             )
 
-            self._generate_library_descriptor(
-                design, sample_name,
-                sample_info['prep']['library_construction_protocol']
-            )
+            # this is the library contruction section. The only required fields
+            # is library_construction_protocol, the other are optional
+            library_descriptor = ET.SubElement(design, 'LIBRARY_DESCRIPTOR')
+            library_name = ET.SubElement(library_descriptor, 'LIBRARY_NAME')
+            library_name.text = self._get_library_name(sample_name)
+
+            lcp = ET.SubElement(library_descriptor,
+                                "LIBRARY_CONSTRUCTION_PROTOCOL")
+            lcp.text = escape(clean_whitespace(
+                sample_prep.pop('library_construction_protocol')))
+
+            # these are not requiered field but present add them in the right
+            # format
+            for field in self.experiment_library_fields:
+                if field in sample_prep:
+                    element = ET.SubElement(library_descriptor, field.upper())
+                    element.text = sample_prep.pop(field)
+                    if field == 'library_layout':
+                        # no idea what SINGLE means
```

Comment:
:cold_sweat: @adamrp EBI whisperer any ideas? Else should we reach to our sequence archivist friends across the pond? Not blocking of course, not the best to have things like this.